Bonsai.Design/ObjectTextVisualizer.cs

```diff
@@ -30,27 +31,90 @@ public class ObjectTextVisualizer : BufferedVisualizer
         /// <inheritdoc/>
         protected override void ShowBuffer(IList<Timestamped<object>> values)
         {
-            if (values.Count > 0)
+            if (values.Count == 0)
+                return;
+
+            var sb = new StringBuilder();
+
+            // Trim old values if bufferSize was reduced
+            while (buffer.Count >= bufferSize)
+                buffer.Dequeue();
+
+            // Add new values to the buffer (and only the ones which might appear)
+            for (int i = Math.Max(0, values.Count - bufferSize); i < values.Count; i++)
             {
-                base.ShowBuffer(values);
-                textBox.Text = string.Join(Environment.NewLine, buffer);
-                textPanel.Invalidate();
+                var rawText = values[i].Value?.ToString() ?? string.Empty;
+                sb.Clear();
+                sb.EnsureCapacity(rawText.Length);
+                MakeDisplayString(rawText, sb);
+                DisplayString(sb.ToString());
             }
+
+            // Update the visual representation of the buffer
+            Debug.Assert(buffer.Count <= bufferSize);
+            sb.Clear();
+            foreach (var line in buffer)
+            {
+                if (sb.Length > 0)
+                    sb.Append(Environment.NewLine);
+                sb.Append(line);
+            }
+            textBox.Text = sb.ToString();
+            textPanel.Invalidate();
         }
 
         /// <inheritdoc/>
         public override void Show(object value)
         {
-            value ??= string.Empty;
-            var text = value.ToString();
-            text = Regex.Replace(text, @"\r|\n", string.Empty);
-            buffer.Enqueue(text);
-            while (buffer.Count > bufferSize)
+            var rawText = value?.ToString() ?? string.Empty;
+            var stringBuilder = new StringBuilder(rawText.Length);
+            MakeDisplayString(rawText, stringBuilder);
+            DisplayString(stringBuilder.ToString());
```

Comment:
Do we need to update the text box and invalidate the text panel here?

Although I agree we probably should just throw here. This code should never be called anyway now that `ShowBuffer` takes care of everything. The idea for buffered visualizers is by default they just call `Show` successively for each value, but since we have a more optimized implementation here this is not used.

Derived classes can just override it again if they need to do something with it.